server/pulp/server/webservices/views/repositories.py

```diff
@@ -0,0 +1,1214 @@
+        :param repo_id: id of requested repository
+        :type  repo_id: str
+
+        :return: Response containing a serialized dict for the created repo.
+        :rtype : django.http.HttpResponse
+        :raises pulp_exceptions.MissingResource if repo cannot be found
+        """
+
+        query_manager = manager_factory.repo_query_manager()
+        repo = query_manager.find_by_id(repo_id)
+
+        if repo is None:
+            raise pulp_exceptions.MissingResource(repo=repo_id)
+
+        repo['_href'] = reverse('repo_resource', kwargs={'repo_id': repo_id})
+        _convert_repo_dates_to_strings(repo)
+
+        details = request.GET.get('details', 'false').lower() == 'true'
+        if request.GET.get('importers', 'false').lower() == 'true' or details:
+            repo = _merge_related_objects(
+                'importers', manager_factory.repo_importer_manager(), (repo,))[0]
+        if request.GET.get('distributors', 'false').lower() == 'true' or details:
+            repo = _merge_related_objects(
+                'distributors', manager_factory.repo_distributor_manager(), (repo,))[0]
+
+        return generate_json_response_with_pulp_encoder(repo)
+
+    @auth_required(authorization.DELETE)
+    def delete(self, request, repo_id):
+        """
+        Dispatch a task to delete a repository.
+
+        :param request: WSGI request object
+        :type  request: django.core.handlers.wsgi.WSGIRequest
+        :param repo_id: id of repository to be removed
+        :type  repo_id: str
+
+        :return: Response containing a serialized dict for the created repo.
+        :rtype : django.http.HttpResponse
+        :raises pulp_exceptions.OperationPostponed: dispatch a task to delete the provided repo
+        """
+
+        # validate
+        manager_factory.repo_query_manager().get_repository(repo_id)
+
+        task_tags = [
+            tags.resource_tag(tags.RESOURCE_REPOSITORY_TYPE, repo_id),
+            tags.action_tag('delete')
+        ]
+        async_result = repo_tasks.delete.apply_async_with_reservation(
+            tags.RESOURCE_REPOSITORY_TYPE, repo_id,
+            [repo_id], tags=task_tags)
+
+        raise pulp_exceptions.OperationPostponed(async_result)
+
+    @auth_required(authorization.UPDATE)
+    @json_body_allow_empty
+    def put(self, request, repo_id):
+        """
+        Update a repository. This call will return synchronously unless a distributor is updated.
+
+        :param request: WSGI request object
+        :type  request: django.core.handlers.wsgi.WSGIRequest
+        :param repo_id: id of repository to be updated
+        :type  repo_id: str
+
+        :return: Response containing a serialized dict for the updated repo.
+        :rtype : django.http.HttpResponse
+        :raises pulp_exceptions.OperationPostponed: if a distributor is updated, dispatch a task
+        """
+
+        delta = request.body_as_json.get('delta', None)
```

Comment:
since body_as_json is an attribute, there wouldn't be a performance difference between the 2, so I think its just a style thing. I find this way slightly more readable, but I don't feel strongly about it.